underlying_functionalities.hpp

```diff
@@ -222,7 +222,7 @@ struct BitWiseRightShiftable : crtp<T, BitWiseRightShiftable>
 template <typename T>
 struct Negatable : crtp<T, Negatable>
 {
-    T operator-() const
+    T negate() const
```

Comment:
Could you please tell me the rationale behind this?

Reply:
I took a look at my code (keep in mind that its over 1 year old and I've learned a great deal about C++ and programming in general during this year) and I honestly can not think of a good reason for that change. The Negetable trait providing the `operator-` is exactly what you would expect it to do and I don't see any good reason to change that...